cve_bin_tool/version_scanner.py

```diff
@@ -140,7 +131,38 @@ def scan_file(self, filename):
             )
         else:
             # Otherwise, use python implementation
-            lines = Strings(filename).parse()
+            s = Strings(filename)
+            lines = s.parse()
+        return lines
+
+    def scan_file(self, filename):
+        """Scans a file to see if it contains any of the target libraries,
+        and whether any of those contain CVEs"""
+
+        self.logger.debug(f"Scanning file: {filename}")
+        self.total_scanned_files += 1
+
+        # Do not try to scan symlinks
+        if os.path.islink(filename):
+            return None
+
+        # Ensure filename is a file
+        if not os.path.isfile(filename):
+            self.logger.warning(f"Invalid file {filename} cannot be scanned")
+            return None
+
+        # check if it's an ELF binary file
+        try:
+            t, output = self.is_executable(filename)
+        except:
+            t = self.is_executable(filename)
+        
+        if not t:
+            return None
```

Comment:
I don't know if this is the best implementation... But, this is what I decided to go with.
Any ideas on how to improve?